docs: add key terms to use case intros/tutorial and what is dvc? docs [SEO] (#1806)
iterative/dvc.org: content/docs/use-cases/versioning-data-and-model-files/index.md

## DVC is not Git!

DVC metafiles such as `dvc.yaml` and `.dvc` files serve as placeholders to track

> the version of data files and directories (among other purposes). They point to

Reviewer: OK I see you're going for "versioning data files" here — thanks for adding the summary of terms in the PR description.

In DVC we try to make an important distinction between tracking and versioning though (the latter mostly left to Git) so let's keep both? May be as simple as

```suggestion
data files and directories for versioning (among other purposes). They point to
```

Or something else (again, don't be afraid to add a sentence somewhere if that doesn't work).
